### PR TITLE
feat(components/ag-grid)!: add support for AG Grid 32

### PR DESCRIPTION
**Pull request by johnhwhite**
[AB#2984635](https://dev.azure.com/blackbaud/f565481a-7bc9-4083-95d5-4f953da6d499/_workitems/edit/2984635) [AB#2994383](https://dev.azure.com/blackbaud/f565481a-7bc9-4083-95d5-4f953da6d499/_workitems/edit/2994383) [AB#2860089](https://dev.azure.com/blackbaud/f565481a-7bc9-4083-95d5-4f953da6d499/_workitems/edit/2860089)

### CODE REVIEW

**Comment by blackbaud-sky-build-user** [comment]
[Storybook preview](https://blackbaud.github.io/skyux-pr-preview/2573/storybook/)

Component Storybooks:

- [a11y-storybook](https://blackbaud.github.io/skyux-pr-preview/2573/storybooks/a11y-storybook/)
- [action-bars-storybook](https://blackbaud.github.io/skyux-pr-preview/2573/storybooks/action-bars-storybook/)
- [ag-grid-storybook](https://blackbaud.github.io/skyux-pr-preview/2573/storybooks/ag-grid-storybook/)
- [angular-tree-component-storybook](https://blackbaud.github.io/skyux-pr-preview/2573/storybooks/angular-tree-component-storybook/)
- [avatar-storybook](https://blackbaud.github.io/skyux-pr-preview/2573/storybooks/avatar-storybook/)
- [colorpicker-storybook](https://blackbaud.github.io/skyux-pr-preview/2573/storybooks/colorpicker-storybook/)
- [core-storybook](https://blackbaud.github.io/skyux-pr-preview/2573/storybooks/core-storybook/)
- [data-manager-storybook](https://blackbaud.github.io/skyux-pr-preview/2573/storybooks/data-manager-storybook/)
- [datetime-storybook](https://blackbaud.github.io/skyux-pr-preview/2573/storybooks/datetime-storybook/)
- [errors-storybook](https://blackbaud.github.io/skyux-pr-preview/2573/storybooks/errors-storybook/)
- [flyout-storybook](https://blackbaud.github.io/skyux-pr-preview/2573/storybooks/flyout-storybook/)
- [forms-storybook](https://blackbaud.github.io/skyux-pr-preview/2573/storybooks/forms-storybook/)
- [help-inline-storybook](https://blackbaud.github.io/skyux-pr-preview/2573/storybooks/help-inline-storybook/)
- [indicators-storybook](https://blackbaud.github.io/skyux-pr-preview/2573/storybooks/indicators-storybook/)
- [inline-form-storybook](https://blackbaud.github.io/skyux-pr-preview/2573/storybooks/inline-form-storybook/)
- [layout-storybook](https://blackbaud.github.io/skyux-pr-preview/2573/storybooks/layout-storybook/)
- [lists-storybook](https://blackbaud.github.io/skyux-pr-preview/2573/storybooks/lists-storybook/)
- [lookup-storybook](https://blackbaud.github.io/skyux-pr-preview/2573/storybooks/lookup-storybook/)
- [modals-storybook](https://blackbaud.github.io/skyux-pr-preview/2573/storybooks/modals-storybook/)
- [navbar-storybook](https://blackbaud.github.io/skyux-pr-preview/2573/storybooks/navbar-storybook/)
- [pages-storybook](https://blackbaud.github.io/skyux-pr-preview/2573/storybooks/pages-storybook/)
- [phone-field-storybook](https://blackbaud.github.io/skyux-pr-preview/2573/storybooks/phone-field-storybook/)
- [popovers-storybook](https://blackbaud.github.io/skyux-pr-preview/2573/storybooks/popovers-storybook/)
- [progress-indicator-storybook](https://blackbaud.github.io/skyux-pr-preview/2573/storybooks/progress-indicator-storybook/)
- [split-view-storybook](https://blackbaud.github.io/skyux-pr-preview/2573/storybooks/split-view-storybook/)
- [storybook](https://blackbaud.github.io/skyux-pr-preview/2573/storybooks/storybook/)
- [tabs-storybook](https://blackbaud.github.io/skyux-pr-preview/2573/storybooks/tabs-storybook/)
- [text-editor-storybook](https://blackbaud.github.io/skyux-pr-preview/2573/storybooks/text-editor-storybook/)
- [theme-storybook](https://blackbaud.github.io/skyux-pr-preview/2573/storybooks/theme-storybook/)
- [tiles-storybook](https://blackbaud.github.io/skyux-pr-preview/2573/storybooks/tiles-storybook/)
- [toast-storybook](https://blackbaud.github.io/skyux-pr-preview/2573/storybooks/toast-storybook/)

Apps:

- [code-examples](https://blackbaud.github.io/skyux-pr-preview/2573/code-examples/)
- [integration](https://blackbaud.github.io/skyux-pr-preview/2573/integration/)
- [playground](https://blackbaud.github.io/skyux-pr-preview/2573/playground/)

File: apps/code-examples/src/app/code-examples/ag-grid/data-entry-grid/basic/demo.component.ts [hunk 1]
```diff
@@ -183,7 +183,7 @@ export class DemoComponent {
     this.searchText = searchText ?? '';
 
     if (this.#gridApi) {
-      this.#gridApi.setQuickFilter(this.searchText);
+      this.#gridApi.updateGridOptions({ quickFilterText: this.searchText });
```
Comment (johnhwhite) [inline apps/code-examples/src/app/code-examples/ag-grid/data-entry-grid/basic/demo.component.ts:186]: This is a schematic change.

File: apps/e2e/ag-grid-storybook/src/app/ag-grid/ag-grid-stories.component.ts [hunk 2]
```diff
@@ -184,7 +179,7 @@ export class AgGridStoriesComponent
           onGridReady: (params) => {
             if (dataSet.id === 'row-delete') {
               params.api.addEventListener(
-                RowNode.EVENT_ROW_SELECTED,
+                'rowSelected',
```
Comment (johnhwhite) [inline apps/e2e/ag-grid-storybook/src/app/ag-grid/ag-grid-stories.component.ts:182]: This is a schematic change.

File: apps/playground/src/app/components/ag-grid/readonly-grid/readonly-grid.component.ts [hunk 3]
```diff
@@ -140,21 +139,18 @@ export class ReadonlyGridComponent implements OnInit {
     this.gridApi = gridReadyEvent.api;
     this.gridApi.sizeColumnsToFit();
     this.gridApi.resetRowHeights();
-    this.gridApi.addEventListener(
-      Events.EVENT_ROW_SELECTED,
```
Comment (johnhwhite) [inline apps/playground/src/app/components/ag-grid/readonly-grid/readonly-grid.component.ts:144]: `Events.EVENT_*` are replaced via the schematic. AG Grid switched to a string union type.

File: libs/components/ag-grid/src/lib/styles/_ag-grid-extra.scss [hunk 4]
```diff
@@ -0,0 +1,644 @@
+@use 'node_modules/ag-grid-community/styles' as ag;
+
+/**
+ * Additional styles based on https://github.com/ag-grid/ag-grid/blob/b32.0.2/community-modules/styles/src/internal/themes/quartz/_index.scss
```
Comment (Blackbaud-SteveBrush) [inline libs/components/ag-grid/src/lib/styles/_ag-grid-extra.scss:4]: Do you have more context around this change?

File: libs/components/ag-grid/src/lib/styles/_base.scss [hunk 5]
```diff
@@ -150,13 +151,19 @@ $modernDarkThemeCompact: map.merge($modernDarkThemeBase, $modernThemeCompact);
 .ag-theme-sky-data-entry-grid-modern-light-compact,
 .ag-theme-sky-data-grid-modern-dark-compact,
 .ag-theme-sky-data-entry-grid-modern-dark-compact {
+  @include ag-grid-extra.ag-grid-extra();
```
Comment (Blackbaud-SteveBrush) [inline libs/components/ag-grid/src/lib/styles/_base.scss:154]: I'm seeing a handful of style changes. Are the new visual tests screenshots desired? Looks like a new row is added underneath the grids...

File: libs/components/ag-grid/src/lib/styles/_ag-grid-extra.scss (hunk 4 above)
Comment (johnhwhite) [reply libs/components/ag-grid/src/lib/styles/_ag-grid-extra.scss:4]: ~~After re-reviewing these changes, I'm dropping them.~~

I pared down these changes to only include the styles we can test. These styles are addressing AG Grid features that our theme was not styling.

File: libs/components/ag-grid/src/lib/styles/_base.scss (hunk 5 above)
Comment (johnhwhite) [reply libs/components/ag-grid/src/lib/styles/_base.scss:154]: Added new visual tests.

**Review by Blackbaud-SteveBrush — DISMISSED** [review]

**Review by Blackbaud-SteveBrush — APPROVED** [review]

File: libs/components/ag-grid/src/lib/styles/_default.scss [hunk 6]
```diff
@@ -53,8 +53,10 @@
           height: 35px;
         }
 
-        &:hover {
-          background-color: variables.$sky-background-color-selected;
+        &:focus-visible,
+        &:hover,
+        &.ag-active-item {
+          background-color: var(--sky-background-color-selected);
```
Comment (johnhwhite) [inline libs/components/ag-grid/src/lib/styles/_default.scss:59]: Fixes AB#2860089

**Pull request merged**